hm_18/hm_018.py

```diff
@@ -0,0 +1,103 @@
+# task_num_1
+class Bot:
+    def __init__(self, name):
+        self.name = name
+
+    def say_name(self):
+        print(self.name)
+
+    def send_message(self, message):
+        print(message)
+
+# task_num_2
+
+
+class TelegramBot(Bot):
+    def __init__(self, name):
+        super().__init__(name)
+        self.url = None
+        self.chat_id = None
+
+    def set_url(self, url):
+        self.url = url
+
+    def set_chat_id(self, chat_id):
+        self.chat_id = chat_id
+
+    def send_message(self, message):
+        print(f"This telegram bot says {message} to chat {self.chat_id} using {self.url}")
+
+# task_num_3
+
+
+class MyStr(str):
+    def __str__(self, *args, **kwargs):
+        return str(self.upper())
```

Comment:
You don't need to transform the result of `self.upper()` to str since `upper` returns a copy of the string converted to uppercase.